ACPs/194-streaming-asynchronous-execution/README.md

```diff
@@ -118,49 +125,123 @@ ACP-103 introduced the following variables for calculating the gas price:
+$$
 
 > [!NOTE]
-> The ACP-103 concept of a leaky gas bucket with capacity $C$ is an emergent phenomenon in SAE.
+> The update rule here assumes that $t_e$ is a timestamp that can track the passage of time both by gas and by wall-clock time. $\frac{g_C}{R}$ MUST NOT be simply rounded. Rather, the gas accumulation should be left as a fraction.
 
-### JSON RPC methods
+$t_e$ is now this block's execution timestamp.
 
-Although asynchronous execution decouples the transactions and receipts recorded by a specific block, APIs MUST NOT alter their behavior to mirror this.
-In particular, the API method `eth_getBlockReceipts` MUST return the receipts corresponding to the block's transactions, not the receipts settled in the block.
+### Handling gas target changes
 
-To achieve this, we first define the _executed_ and _settled_ states for blocks.
-A block is initially _accepted_ by consensus, enqueuing its tranche of transactions.
-Once the last transaction in the tranche has been executed, the block itself is considered to be executed.
-When the same transaction is settled then so too is the block considered settled.
+When a block is produced which modifies $T$, both the consensus thread and the execution thread will update to the modified $T$ after their own handling of the block.
 
-#### Named blocks
+For example, restrictions of the queue size MUST be calculated based on the parent block's $T$.
 
-The Ethereum Mainnet APIs allow for retrieving blocks by named parameters that the API server resolves based on their consensus mechanism.
-Other than the _earliest_ (genesis) named block, which MUST be interpreted in the same manner, all other named blocks are mapped to SAE in terms of the execution status of all block transactions and MUST be interpreted as follows:
+Similarly, the time spent executing a block MUST be calculated based on the parent block's $T$.
 
- * _pending_: the most recently _accepted_ block;
- * _latest_: the block that was most recently _executed_;
- * _safe_ and _finalized_: the block that was most recently _settled_.
+### Block settlement
+
+For a _proposed_ block which includes timestamp $t_b$, all ancestors whose execution timestamp $t_e$ is $t_e + \tau \leq t_b$ are considered settled.
+
+The _proposed_ block must include the `stateRoot` produced by the execution of the most recently settled block.
+
+For any _newly_ settled blocks, the _proposed_ block must include all execution artifacts:
+- `receiptsRoot`
+- `logsBloom`
+- `gasUsed`
 
 > [!NOTE]
-> The finality guarantees of Snowman consensus remove any distinction between _safe_ and _finalized_. 
-> Furthermore, the _latest_ block is not at risk of re-org, only of a negligible risk of data corruption local to the API node.
+> If the block executor has fallen behind, the node may not be able to determine precisely which ancestors should be considered settled. If this occurs, validators MUST allow the block executor to catch up prior to deciding the block's validity.
```

Comment:
Do we want to say something similar about block proposers wishing to propose a block at a given timestamp? It's implicit in the validation logic, but is it necessary to call it out?

Reply:
Your call

Reply:
Let's leave it as is. Adding more might be explicit but could also cause confusion if someone things they may be different. Since `Block.Verify()` always enforces `vm.BuildBLock()`, this is safe IMO. Resolving.